drivers/display/ssd16xx.c

```diff
@@ -62,42 +62,38 @@ struct ssd16xx_data {
 	u8_t scan_mode;
 };
 
+#if defined(DT_INST_0_SOLOMON_SSD16XXFB_LUT_INITIAL)
 static u8_t ssd16xx_lut_initial[] = DT_INST_0_SOLOMON_SSD16XXFB_LUT_INITIAL;
+#endif
+#if defined(DT_INST_0_SOLOMON_SSD16XXFB_LUT_DEFAULT)
 static u8_t ssd16xx_lut_default[] = DT_INST_0_SOLOMON_SSD16XXFB_LUT_DEFAULT;
+#endif
 #if defined(DT_INST_0_SOLOMON_SSD16XXFB_SOFTSTART)
 static u8_t ssd16xx_softstart[] = DT_INST_0_SOLOMON_SSD16XXFB_SOFTSTART;
 #endif
 static u8_t ssd16xx_gdv[] = DT_INST_0_SOLOMON_SSD16XXFB_GDV;
 static u8_t ssd16xx_sdv[] = DT_INST_0_SOLOMON_SSD16XXFB_SDV;
 
-#ifndef DT_INST_0_SOLOMON_SSD16XXFB_LUT_INITIAL
-#error "No initial waveform look up table (LUT) selected!"
-#endif
-
-#ifndef DT_INST_0_SOLOMON_SSD16XXFB_LUT_DEFAULT
-#error "No default waveform look up table (LUT) selected!"
-#endif
+#define SSD16XX_DEFAULT_TR_VALUE	25U
+#define SSD16XX_TR_SCALE_FACTOR		256U
 
 static inline int ssd16xx_write_cmd(struct ssd16xx_data *driver,
 				    u8_t cmd, u8_t *data, size_t len)
 {
-	int err;
 	struct spi_buf buf = {.buf = &cmd, .len = sizeof(cmd)};
 	struct spi_buf_set buf_set = {.buffers = &buf, .count = 1};
 
 	gpio_pin_write(driver->dc, SSD16XX_DC_PIN, 0);
-	err = spi_write(driver->spi_dev, &driver->spi_config, &buf_set);
-	if (err < 0) {
-		return err;
+	if (spi_write(driver->spi_dev, &driver->spi_config, &buf_set)) {
+		return -EIO;
```

Comment:
What is the reason to change this and not just return the actual error code to the caller?

Reply:
Do not forward minor spi_write return value and
**save few lines of code** (`@retval 0 If successful, negative errno code otherwise.`)
Treat undefined positive return value as an error.

Reply:
> edit: more precise, unspecified return value

You have seen them, if this the case sound like a bug in a SPI driver?

Reply:
oh no, it is just unspecified, but we have it on many places. The change above is more optical.